phpunit/DbTestCase.php

```diff
@@ -200,6 +200,7 @@ protected function createItem($itemtype, $input, $skip_fields = []): CommonDBTM
     {
         $item = new $itemtype();
         $id = $item->add($input);
+        $id = $item->getID();
```

Comment:
Is this change intentional ? seems weird

Reply:
Yes, because the add() function returns the value of the column named 'id', whereas getID() uses the column from getIndexName(), like getFromDB().
In most cases, getIndexName() returns 'id', but for the 'glpi_ticketsatisfactions' table, it's 'tickets_id'.

Reply:
I guess we should keep the `id` field here. If you need the field from the `getId()` method, you'll have to retrieve it specifically.

Reply:
I did this so that the L121 checkInput() test would pass, since getFromDB() uses getIndexName()

Reply:
Unfortunately, getFromDByCrit() also uses getFromDB() 😢 

![image](https://github.com/user-attachments/assets/4aab8904-a261-437d-9b46-74cf5c05deff)

Reply:
It uses `getFromDB()` with the `id` field - that sounds like a mistake and can lead to errors... All that stuff is always a nightmare; let's discuss about that when all core team will be present.

Reply:
I've opened a PR to fix CommonDBTM, see https://github.com/glpi-project/glpi/pull/17696